consensus/vbft/service.go

```diff
@@ -229,6 +229,25 @@ func (self *Server) handleBlockPersistCompleted(block *types.Block) {
 	}
 }
 
+func (self *Server) SubmitBlock(blkNum uint32, stateRoot common.Uint256) {
+	cMsgs := self.msgPool.GetBlockSubmitMsgNums(blkNum)
+	m := len(self.config.Peers) - (len(self.config.Peers)*3)/7
+	if len(cMsgs) < m {
+		return
+	}
+	for _, msg := range cMsgs {
+		c := msg.(*blockSubmitMsg)
+		if c != nil {
+			if c.BlockStateRoot != stateRoot {
```

Comment:
why not check it before `AddMsg`?

Reply:
there are some concurrency problem when  call GetExecMerkleRoot() function,so get merkroot in same goroutine.